src/iceberg/expression/aggregate.cc

```diff
@@ -152,6 +285,11 @@ Result<Literal> CountAggregate::Evaluate(const StructLike& data) const {
   return CountFor(data).transform([](int64_t count) { return Literal::Long(count); });
 }
 
+Result<Literal> CountAggregate::Evaluate(const DataFile& file) const {
+  ICEBERG_ASSIGN_OR_RAISE(auto count, CountFor(file));
+  return Literal::Long(count);
```

Comment:
```suggestion
  return CountFor(file).transform(Literal::Long);
```
Please also help simplify `CountAggregate::Evaluate(const StructLike& data) const` above.